jupyterhub_moss/spawner.py

```diff
@@ -101,58 +173,74 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.options_form = self.create_options_form
 
-    async def _get_slurm_info(self):
-        """Returns information about partitions from slurm"""
+    async def _get_slurm_info_resources(self):
+        """
+        Retrieves information about resources in partitions from slurm
+        1. executes slurm_info_cmd
+        2. parses output with slurm_info_resources
+        """
+        # Execute given slurm info command
         subvars = self.get_req_subvars()
         cmd = " ".join(
             (
                 format_template(self.exec_prefix, **subvars),
                 format_template(self.slurm_info_cmd, **subvars),
             )
         )
-        self.log.info("Slurm info command: %s", cmd)
-        state = await self.run_command(cmd)
-        slurm_info = defaultdict(lambda: {"nodes": 0, "idle": 0, "max_mem": 0})
-        for line in state.splitlines():
-            partition, state, memory = line.split()
-            info = slurm_info[partition]
-            info["nodes"] += 1
-            if state == "idle":
-                info["idle"] += 1
-            info["max_mem"] = max(info["max_mem"], int(memory))
-        return slurm_info
+        self.log.debug("Slurm info command: %s", cmd)
+        out = await self.run_command(cmd)
+
+        # Parse command output
+        resources_display, resources_info = self._slurm_info_resources(out)
```

Comment:
Unless I missed something, his should be, right?:
```suggestion
        resources_display, resources_info = self.slurm_info_resources(out)
```

Reply:
indeed, you are right :+1: